homeassistant/components/doods/image_processing.py

```diff
@@ -0,0 +1,366 @@
+    (left, right, top, bottom) = (
+        xmin * img_width,
+        xmax * img_width,
+        ymin * img_height,
+        ymax * img_height,
+    )
+    draw.line(
+        [(left, top), (left, bottom), (right, bottom), (right, top), (left, top)],
+        width=5,
+        fill=color,
+    )
+    if text:
+        draw.text((left, abs(top - 15)), text, fill=color)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Doods client."""
+    url = config[CONF_URL]
+    auth_key = config[CONF_AUTH_KEY]
+    detector_name = config[CONF_DETECTOR]
+
+    doods = PyDOODS(url, auth_key)
+    response = doods.get_detectors()
+    if not isinstance(response, dict):
+        _LOGGER.warning("Could not connect to doods server: %s", url)
+        return
+
+    detector = {}
+    for server_detector in response["detectors"]:
+        if server_detector["name"] == detector_name:
+            detector = server_detector
+            break
+
+    if not detector:
+        _LOGGER.warning(
+            "Detector %s is not supported by doods server %s", detector_name, url
+        )
+        return
+
+    entities = []
+    for camera in config[CONF_SOURCE]:
+        entities.append(
+            Doods(
+                hass,
+                camera[CONF_ENTITY_ID],
+                camera.get(CONF_NAME),
+                doods,
+                detector,
+                config,
+            )
+        )
+    add_entities(entities)
+
+
+class Doods(ImageProcessingEntity):
+    """Doods image processing service client."""
+
+    def __init__(self, hass, camera_entity, name, doods, detector, config):
+        """Initialize the DOODS entity."""
+        self.hass = hass
+        self._camera_entity = camera_entity
+        if name:
+            self._name = name
+        else:
+            self._name = "Doods {0}".format(split_entity_id(camera_entity)[1])
+        self._doods = doods
+        self._file_out = config.get(CONF_FILE_OUT)
+
+        # detector config and aspect ratio
+        if detector["width"] and detector["height"]:
+            self._width = detector["width"]
+            self._height = detector["height"]
+            self._aspect = self._width / self._height
+
+        # the base confidence
+        dconfig = {}
+        confidence = config.get(CONF_CONFIDENCE)
+        if not confidence:
+            confidence = 0
+
+        # handle labels and specific detection areas
+        labels = config.get(CONF_LABELS)
+        self._label_areas = {}
+        for label in labels:
+            if isinstance(label, dict):
+                label_name = label.get(CONF_NAME)
+                if label_name not in detector["labels"] and label_name != "*":
+                    _LOGGER.warning("Detector does not support label %s", label_name)
+                    continue
+
+                # Label Confidence
+                label_confidence = label.get(CONF_CONFIDENCE)
+                if not label_confidence:
```

Comment:
Is this checking for `None` or `0`?

Reply:
I realized it had a default. Removed check.